java/Anton-Kozyk/src/com/zoolatech/lecture6/tasks/_3/UserService.java

```diff
@@ -0,0 +1,24 @@
+package com.zoolatech.lecture6.tasks._3;
+
+public class UserService {
+    private Cache cache = new Cache();
+    private Table userTable = new Table();
+
+    UserRepository userRepository = new UserRepository(cache, userTable);
+
+    public void addUser(User user) {
+        userTable.addToUserList(user);
+    }
+
+    public void addToCache(User user) {
+        cache.addToCache(user);
+    }
+
+    public String findUserEmail(String id) {
+        try {
+            return userRepository.findUserEmail(id);
+        } catch (UserMissingException e) {
+            return e.getLocalizedMessage();
```

Comment:
it's better to not return the exception message as the method result since it can be mixed up with valid return values (email)

you can simply print the message to the output